api/envoy/extensions/filters/network/mysql_proxy/v3/mysql_proxy.proto

```diff
@@ -15,15 +18,71 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
 // MySQL Proxy :ref:`configuration overview <config_network_filters_mysql_proxy>`.
 // [#extension: envoy.filters.network.mysql_proxy]
 
+// Auth information of MySQL.
+message AuthInfo {
+  config.core.v3.DataSource auth_username = 1;
+
+  config.core.v3.DataSource auth_password = 2;
+}
+
+// [#next-free-field: 6]
 message MySQLProxy {
   option (udpa.annotations.versioning).previous_message_type =
       "envoy.config.filter.network.mysql_proxy.v1alpha1.MySQLProxy";
 
+  message Route {
+    // String database name that must exact match with client connection db name
+    string database = 1 [(validate.rules).string = {min_len: 1}];
+
+    // Upstream cluster to forward the command to.
+    string cluster = 2 [(validate.rules).string = {min_len: 1}];
+  }
+
   // The human readable prefix to use when emitting :ref:`statistics
   // <config_network_filters_mysql_proxy_stats>`.
   string stat_prefix = 1 [(validate.rules).string = {min_len: 1}];
 
   // [#not-implemented-hide:] The optional path to use for writing MySQL access logs.
   // If the access log field is empty, access logs will not be written.
   string access_log = 2;
+
+  // [#not-implemented-hide:] enable feature of separate authentication of downstream clients and upstream server. When feature_enable is true, the filter is a terminal filter, it's config should put at the last of filter chain, when feature_enable is false, the filter is not a terminal filter, it's config can not put at the last of filter chain. Default is false.
```

Comment:
Needs to be wrapped and have a better name than "feature_enable"; maybe `enable_protocol_management` or something like that to indicate you're now basically the equivalent of HCM.

Reply:
fixed.